src/storage/mvcc/reader/reader.rs

```diff
@@ -28,6 +30,13 @@ pub enum TxnCommitRecord {
     OverlappedRollback { commit_ts: TimeStamp },
 }
 
+#[derive(Clone, Debug)]
+pub struct OverlappedWrite {
+    pub write: Write,
+    /// GC fence for `overlapped_write`. PTAL at `txn_types::Write::gc_fence`.
+    pub gc_fence: TimeStamp,
```

Comment:
It looks redundant to add the `gc_fence` field here, since `Write` already has the field.

Reply:
It's something different. It's actually the "potential" gc_fence to be write.

Reply:
In fact we can merge them into one. I have updated code for this.